File: C#/TddApi/NavigationApp/s19.refactor_navigator/Navigator.cs
﻿using System;

namespace NavigationApp.s19.refactor_navigator
{
    public class Navigator
    {
        DistanceProvider distProvider;
        Location startingPoint;
        Location destination;

        public Navigator(DistanceProvider distService)
        {
            distProvider = distService;
        }

        public Distance getDistanceFromDestination()
        {
            return distProvider.getDistance(startingPoint, destination);
        }

        public void setDestination(Location dest)
        {
            destination = dest;
        }

        public void setStartingPoint(Location loc)
        {
            startingPoint = loc;
        }
    }
}

File: Python/s20_mock_provider/navigator.py
class Navigator:
    def __init__(self, dist_provider):
        self.dist_provider = dist_provider
        self.starting_point = None
        self.destination = None

    def get_distance_from_destination(self):
        return self.dist_provider.get_distance(self.starting_point, self.destination)

    def set_destination(self, dest):
        self.destination = dest

    def set_start_point(self, startingPoint):
        self.starting_point = startingPoint
